shell/platform/embedder/embedder.h

```diff
@@ -434,6 +435,52 @@ typedef struct {
   BoolPresentInfoCallback present_with_info;
 } FlutterOpenGLRendererConfig;
 
+/// Handle for the MTLDevice.
+typedef const void* FlutterMetalDevice;
+
+/// Handle for the MTLCommandQueue.
+typedef const void* FlutterMetalCommandQueue;
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterMetalTexture).
+  size_t struct_size;
+  /// Embedder provided unique identifier to the texture buffer. Given that the
+  /// `texture` handle is passed to the engine to render to, the texture buffer
+  /// is itseld owned by the embedder. This `texture_id` is then also given to
+  /// the embedder in the present callback.
+  int64_t texture_id;
+  /// Handle to the MTLTexture that is owned by the embedder. Engine will render
+  /// the frame into this texture.
+  void* texture;
+} FlutterMetalTexture;
+
+/// Callback for when a metal texture is requested.
+typedef void (*FlutterMetalTextureCallback)(
+    void* /* user data */,
+    const FlutterFrameInfo* /* frame info */,
+    FlutterMetalTexture* /* texture out */);
+
+/// Callback for when a metal texture is presented. The texture_id here
+/// corresponds to the texture_id provided by the embedder in the
+/// `FlutterMetalTextureCallback` callback.
+typedef bool (*FlutterMetalPresentCallback)(void* /* user data */,
+                                            int64_t /* texture id */);
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterMetalRendererConfig).
+  size_t struct_size;
+  /// Handle for the MTLDevice.
+  FlutterMetalDevice device;
+  /// Handle for the MTLCommandQueue.
+  FlutterMetalCommandQueue command_queue;
```

Comment:
`present_command_queue` for clarity.

Reply:
I forget, does the embedder need to own a reference to the main command queue? Or, can the engine create a command queue on behalf of the embedder?

Reply:
I think this provides the embedder with most flexibility since it might want to time when the commit needs to happen, see: https://github.com/flutter/engine/pull/22606/files#diff-112f37d852d26e9e2c1040c0f0ab516a94450e5d48c37338612bd254040881d3R85. I will rename to `present_command_queue` for clarity.